src/containers/CreatePipelineRun/YAMLEditor.js

```diff
@@ -0,0 +1,245 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { useIntl } from 'react-intl';
+import { ALL_NAMESPACES, urls } from '@tektoncd/dashboard-utils';
+import {
+  Button,
+  Form,
+  FormGroup,
+  InlineNotification
+} from 'carbon-components-react';
+import yaml from 'js-yaml';
+import React, { useState } from 'react';
+import CodeMirror from '@uiw/react-codemirror';
+import { StreamLanguage } from '@codemirror/language';
+import { yaml as cmyaml } from '@codemirror/legacy-modes/mode/yaml';
+import { useNavigate } from 'react-router-dom-v5-compat';
+import { createPipelineRunRaw, useSelectedNamespace } from '../../api';
+
+const initialState = {
+  creating: false,
+  pipelineError: false,
+  submitError: '',
+  validationError: false,
+  validationErrorMessage: '',
+  code: ''
+};
+
+export function CreateYAMLEditor() {
+  const intl = useIntl();
+  const navigate = useNavigate();
+  const { selectedNamespace } = useSelectedNamespace();
+
+  const [
+    {
+      creating,
+      pipelineError,
+      submitError,
+      validationError,
+      validationErrorMessage,
+      code
+    },
+    setState
+  ] = useState({
+    ...initialState
+  });
+
+  function validateNamespace(obj) {
+    if (!obj?.metadata?.namespace) {
+      return {
+        valid: false,
+        message: intl.formatMessage({
+          id: 'dashboard.createRun.invalidNamespace',
+          defaultMessage: 'Namespace cannot be empty'
+        })
+      };
+    }
+    return null;
+  }
+
+  function validateEmptyYaml() {
+    if (!code || code === '') {
+      return {
+        valid: false,
+        message: intl.formatMessage({
+          id: 'dashboard.createPipelineRun.invalidPipelineRun',
+          defaultMessage: 'PipelineRun cannot be empty'
+        })
+      };
+    }
+    return null;
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    // Check form validation
+    let validationResult = validateEmptyYaml();
+    if (validationResult && !validationResult.valid) {
+      setState(state => ({
+        ...state,
+        validationError: true,
+        validationErrorMessage: validationResult.message
+      }));
+      return;
+    }
+
+    let obj;
+    try {
+      obj = yaml.load(code);
+    } catch (e) {
+      setState(state => ({
+        ...state,
+        validationError: true,
+        validationErrorMessage: e.message
+      }));
+      return;
+    }
+
+    validationResult = validateNamespace(obj);
+    if (validationResult && !validationResult.valid) {
+      setState(state => ({
+        ...state,
+        validationError: true,
+        validationErrorMessage: validationResult.message
+      }));
+      return;
+    }
+
+    setState(state => ({ ...state, creating: true }));
+    const nspace = obj?.metadata?.namespace;
```

Comment:
nit/naming: `namespace` for consistency
also means you can use shorthand properties in objects, e.g.
`{ namespace: namespace }` becomes `{ namespace }`